GameOfLife.java

```diff
@@ -50,63 +52,150 @@ public static void play(String fileName) {
 			board = evolve(board);
 		}
 	}
-	
-	// Reads the initial board configuration from the file whose name is fileName, uses the data
-	// to construct and populate a 2D array that represents the game board, and returns this array.
-	// Live and dead cells are represented by 1 and 0, respectively. The constructed board has 2 extra
-	// rows and 2 extra columns, containing zeros. These are the top and the bottom row, and the leftmost
-	// and the rightmost columns. Thus the actual board is surrounded by a "frame" of zeros. You can think
-	// of this frame as representing the infinite number of dead cells that exist in every direction.
-	// This function assumes that the input file contains valid data, and does no input testing.
+
+	// Reads the initial board configuration from the file whose name is fileName,
+	// uses the data
+	// to construct and populate a 2D array that represents the game board, and
+	// returns this array.
+	// Live and dead cells are represented by 1 and 0, respectively. The constructed
+	// board has 2 extra
+	// rows and 2 extra columns, containing zeros. These are the top and the bottom
+	// row, and the leftmost
+	// and the rightmost columns. Thus the actual board is surrounded by a "frame"
+	// of zeros. You can think
+	// of this frame as representing the infinite number of dead cells that exist in
+	// every direction.
+	// This function assumes that the input file contains valid data, and does no
+	// input testing.
 	public static int[][] read(String fileName) {
 		In in = new In(fileName); // Constructs an In object for reading the input file
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
-		//// Replace the following statement with your code.
-		return null;
+		int indexRows = 1;
+
+		while (!in.isEmpty()) {
+			String line = in.readLine();
+			if (line == "") {
+				for (int i = 1; i < cols; i++) {
+					board[indexRows][i] = 0;
+				}
+				indexRows++;
+			}
+			if (!(line == "")) {
+				for (int i = 1; i - 1 < line.length(); i++) {
```

Comment:
no need this if. If the line is empty, then line.length() will be 0 and the loop will not run